src/main/java/org/apache/commons/lang3/time/StopWatch.java

```diff
@@ -671,4 +686,622 @@ public void unsplit() {
+        return (firstArg, secondArg) -> {
+            resumeOrStartStopWatch();
+            try {
+                return predicate.test(firstArg, secondArg);
+            } finally {
+                suspend();
+            }
+        };
+    }
+
+    /**
+     * Take the time of the execution of a given {@link FailableBiPredicate}.
+     *
+     * <p>
+     * <em>Take the time of given {@link FailableBiPredicate}</em>
+     * </p>
+     * <pre>{@code
+     * final StopWatch watch = StopWatch.create();
+     * FailableBiPredicate<String, String> predicate = Objects::equals;
+     * boolean result = watch.test(predicate).test("A", "B");
+     * }</pre>
+     *
+     * <p>
+     *  <em>Note:</em>
+     * </p>
+     * <ul>
+     *     <li>Be careful if you pass a {@link FailableBiPredicate} to other methods, as the wrapped instance depends on
+     *         the state of used {@linkplain StopWatch}</li>
+     * </ul>
+     *
+     * @param predicate the predicate those application should be measured
+     * @return the given predicate prepared to take time if applied
+     * @throws IllegalStateException if the {@link StopWatch} is not stopped or suspended
+     * @param <T> the type of the first argument to the predicate
+     * @param <U> the type of the second argument to the predicate
+     * @param <E> The kind of thrown exception or error
+     * @since 3.17
+     */
+    public <T, U, E extends Throwable> FailableBiPredicate<T, U, E> test(final FailableBiPredicate<T, U, E> predicate) {
+        return (firstArg, secondArg) -> {
+            resumeOrStartStopWatch();
+            try {
+                return predicate.test(firstArg, secondArg);
+            } finally {
+                suspend();
+            }
+        };
+    }
+
+    /**
+     * Take the time of the execution of a given {@link FailablePredicate}.
+     *
+     * <p>
+     * <em>Take the time of given {@link FailablePredicate}</em>
+     * </p>
+     * <pre>{@code
+     * final StopWatch watch = StopWatch.create();
+     * FailablePredicate<String> predicate = false;
+     * boolean result = watch.test(predicate).test("A");
+     * }</pre>
+     *
+     * <p>
+     *  <em>Note:</em>
+     * </p>
+     *  <ul>
+     *      <li>Be careful if you pass a {@link FailablePredicate} to other methods, as the wrapped instance depends on
+     *          the state of used {@linkplain StopWatch}</li>
+     *  </ul>
+     *
+     * @param predicate the predicate those application should be measured
+     * @return the given predicate prepared to take time if applied
+     * @throws IllegalStateException if the {@link StopWatch} is not stopped or suspended
+     * @param <T> the type of the argument to the predicate
+     * @param <E> The kind of thrown exception or error
+     * @since 3.17
+     */
+    public <T, E extends Throwable> FailablePredicate<T, E> test(final FailablePredicate<T, E> predicate) {
+        resumeOrStartStopWatch();
```

Comment:
Tests should catch this.

Reply:
Why was this one marked as resolved? Test was not added nor the bug fixed.

Reply:
You are right. Sorry, I didn't see this